dialects/linux/dproc.c

```diff
@@ -1395,6 +1397,57 @@ process_id(idp, idpl, cmd, uid, pid, ppid, pgid, tid, tcmd)
 	return(0);
 }
 
+/* compare namespace of this lsof proces and the target process */
+
+int
+compare_namespace(ns)
+        char *ns;                /* namespace */
+{
+        char nspath_self[NS_PATH_LENGTH], nspath_target[NS_PATH_LENGTH];
+        char ns_self[NS_LENGTH], ns_target[NS_LENGTH];
+        ssize_t n_self, n_target;
+        int ret;
+
+        ret = snprintf(nspath_self, NS_PATH_LENGTH, "/proc/self/ns/%s", ns);
+        if (ret >= NS_PATH_LENGTH || ret <= 0)
+	    return -1;
+
+        n_self = readlink(nspath_self, ns_self, NS_LENGTH);
```

Comment:
Instead of using `NS_LENGTH`, you can use `sizeof ns_self`.
 I wonder which one is better.

Reply:
Helpful, sizeof is better because it's more readable and alterable.